### ISSUE TITLE
Various build warnings (treated as errors): error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]

### ISSUE THREAD

**Issue by dholbert** [rr-debugger/rr#3386]
When building rr with Ubuntu clang version 15.0.2-1 on Ubuntu 22.10, I get the following warnings (treated as errors):
```
rr/src/AddressSpace.cc:212:24: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
      mapped_file_stat(move(mapped_file_stat)),
                       ^
                       std::
rr/src/AddressSpace.cc:214:31: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
      monitored_shared_memory(move(monitored)),
                              ^
                              std::
rr/src/AddressSpace.cc:777:57: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
  map_and_coalesce(t, m, actual_recorded_map, emu_file, move(mapped_file_stat),
                                                        ^
                                                        std::
rr/src/AddressSpace.cc:778:20: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
                   move(local_addr), move(monitored));
                   ^
                   std::
rr/src/AddressSpace.cc:778:38: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
                   move(local_addr), move(monitored));
                                     ^
                                     std::
rr/src/AddressSpace.cc:893:17: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
    Mapping m = move(mm);
                ^
                std::
rr/src/AddressSpace.cc:920:11: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
          move(monitored));
          ^
          std::
rr/src/AddressSpace.cc:1298:17: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
    Mapping m = move(mm);
                ^
                std::
rr/src/AddressSpace.cc:1310:39: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
                        m.local_addr, move(monitored));
                                      ^
                                      std::
rr/src/AddressSpace.cc:1543:23: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
    Mapping mapping = move(mm);
                      ^
                      std::
rr/src/AddressSpace.cc:2066:45: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
      m, Mapping(m, recorded_map, emu_file, move(mapped_file_stat), local_addr,
                                            ^
                                            std::
rr/src/AddressSpace.cc:2067:18: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
                 move(monitored))));
                 ^
                 std::
rr/src/Dwarf.cc:201:43: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
    abbrevs.insert(make_pair(abbrev_code, move(abbrev)));
                                          ^
                                          std::
rr/src/Dwarf.cc:218:36: error: unqualified call to 'std::move' [-Werror,-Wunqualified-std-cast-call]
  abbrevs.insert(make_pair(offset, move(set)));
                                   ^
                                   std::
```

This building is new to me so I'm not sure about the reasoning behind it, but it looks like it's addressed by adding `std::` prefix on all `move` invocations.

(This is needed even despite `using namespace std;` at the top of the file -- maybe for disambiguation, in case a project ever gets a `move` symbol from some other namespace? Not sure.)

**Comment by dholbert** [comment]
(The above are just the first instances of this issue that I hit, which I patched by hand.  There are additional warnings further into the build for `ExportImportCheckpoints.cc` and `GdbServer.cc` and `GdbExpression.cc` and `GdbConnection.cc`, at least.)

**Comment by dholbert** [comment]
Fix posted in #3387,

**Comment by dholbert** [comment]
(I confirmed that I can build successfully with that pull request's commit.)
